packages/wrangler/src/kv.ts

```diff
@@ -228,45 +227,26 @@ export function getNamespaceId(
   // begin pre-flight checks
 
   // `--binding` is only valid if there's a wrangler configuration.
-  if (binding && !configOrEnv) {
+  if (binding && !config) {
     throw new Error("--binding specified, but no config file was found.");
   }
 
   // there's no config. abort here
-  if (!configOrEnv) {
+  if (!config) {
     throw new Error(
       "Failed to find a config file.\n" +
         "Either use --namespace-id to upload directly or create a configuration file with a binding."
     );
   }
 
-  // they want to use an environment, actually
-  if (env) {
-    if (!("env" in configOrEnv) || !configOrEnv.env[env]) {
-      throw new Error(
-        `Failed to find environment "${env}" in configuration file!`
-      );
-    }
-    return getNamespaceId(
-      {
-        binding,
-        "namespace-id": namespaceId,
-        preview,
-      },
-      configOrEnv.env[env]
-    );
-  }
-
```

Comment:
This is an example of how this PR helps to simplify downstream code.
The `getNamespaceId()` function no longer needs to know about how environment inheritance works in the configuration.